### PR TITLE
refactor(utils): consolidate string and option handling

### PR DESCRIPTION
**Pull request by lxy-9602**
### Purpose

This PR consolidates duplicated string and option handling into the common utility classes.

The main changes are:

- Align `StringUtils` edge-case behavior with the corresponding Java implementation, including replace, trim, and starts-with operations.
- Add ASCII-based case conversion and case-insensitive comparison helpers.
- Add `OptionsUtils::GetNonEmptyValueFromMap` and improve prefixed option extraction.
- Replace duplicated string and option handling across REST, filesystem, schema validation, variant, and system-table code.
- Reuse `SpecialFields::CompleteSequenceAndValueKindField` when constructing the postpone-bucket write schema.
- Remove redundant `parquet.read.enable-pre-buffer=false` injection from compaction paths. Compaction always enables framework prefetch, which already disables Parquet pre-buffering, so mutating the format options is unnecessary.

### Tests
Added or updated unit coverage for:

- `StringUtils` edge cases and case-insensitive comparison.
- `OptionsUtils` non-empty values and prefixed option extraction.
- Case-insensitive local filesystem schemes.
- Case-insensitive Parquet format validation.
- Sequence snapshot property parsing.

### Generative AI tooling
Generated-by: OpenAI Codex (GPT-5)

### CODE REVIEW

File: src/paimon/common/utils/options_utils.h [hunk 1]
```diff
@@ -89,13 +89,29 @@ class OptionsUtils {
...
+        }
+        if (value.value().empty()) {
+            return Status::Invalid(fmt::format("value for key {} must not be empty", key));
+        }
+        return value.value();
+    }
+
     /// Fetch options with specific prefix and remove prefix for key.
+    ///
+    /// If `ignore_empty_key` is true, an option whose key equals `prefix` is ignored.
```
Comment (zjw1111) [inline src/paimon/common/utils/options_utils.h]: use doxygen format

**Review by zjw1111 — APPROVED** [review]
+1

**Review by SteNicholas — APPROVED** [review]
LGTM.
